playbook/run_playbook/main.yml

```diff
@@ -0,0 +1,135 @@
+        vpc_subnet_id: "{{ build_subnet_id }}"
+        count: 1
+        termination_protection: no
+        wait: yes
+        wait_timeout: 500
+        user_data: "{{ lookup('file', 'user_data.sh') }}"
+        instance_profile_name: "bb-{{ env }}-ephemeral-profile"
+
+        # tags to instance_tags
+        instance_tags:
+          Name: "bb-{{ env }}-{{ azone }}-run-playbook"
+          Stack: "BB-{{ env|upper }}-{{ cf_app_azone|upper }}-{{ cf_app_tag_key_layer|upper }}-ANSIBLE"
+          Business: "{{ cf_tags_business }}"
+          Application: "{{ cf_tags_application }}"
+          Environment: "{{ env|upper }}"
+          Function: "{{ cf_app_tag_key_layer }}-AppServer"
+          Layer:  "{{ cf_app_tag_key_layer|upper }}"
+          ami_name: "BB-{{ env|upper }}-{{ cf_app_tag_key_layer|upper }}"
+          region: "{{ cf_region }}"
+          Managed: "BB-MANAGED-{{ env|upper }}"
+          State: "ami-build"
+      register: app_ec2_output
+
+    - name: "Add new instance to inventory ( {{ app_ec2_output.instances[0]['private_ip'] }} ) "
+      add_host:
+        name: "{{ app_ec2_output.instances[0]['private_ip'] }}"
+        groups: run_servers
+
+    - name: Wait 180 seconds, but only start checking after 60 seconds
+      wait_for_connection:
+        delay: 60
+        timeout: 180
+
+    - name: "Wait for SSH to become available"
+      wait_for:
+        port: 22
+        host: "{{ app_ec2_output.instances[0]['private_ip'] }}"
+        delay: 5
+
+
+- name: Provision EC2 instance
+  hosts: run_servers
+  remote_user: ec2-user
+  gather_facts: no
+  vars:
+    ansible_ssh_pipelining: no
+
+    roles:
+    - ../../roles/base_patch
+
+
+- name: "playbook_debug_fail before nested playbook if set = 1"
+  hosts: localhost
+  connection: local
+
+  tasks:
+    - name: "playbook_debug_fail before nested playbook if set = 1"
+      fail: msg="Fail out early to skip remaining tasks and leave instance running."
+      when: playbook_debug_fail == "1"
+
+
+# Nested playbook import:
+
+- import_playbook: "../{{ playbook }}/main.yml"
+  ignore_errors: yes
+
+
+- name: "playbook_debug_fail after nested playbook if set = 2"
+  hosts: localhost
+  connection: local
+
+  tasks:
+    - name: "playbook_debug_fail after nested playbook if set = 2"
+      fail: msg="Fail out early to skip remaining tasks and leave instance running."
+      when: playbook_debug_fail == "2"
+
+
+- name: Decommission EC2 instance
+  hosts: run_servers
+  remote_user: ec2-user
+  gather_facts: no
+  vars:
+    ansible_ssh_pipelining: no
+
+  post_tasks:
+    - name: Gather instance metadata
+      ec2_metadata_facts:
+
+    - name: "Stop the EC2 instance"
+      ec2:
+        instance_ids: '{{ ansible_ec2_instance_id }}'
+        region: '{{ ansible_ec2_placement_region }}'
+        state: stopped
+        wait: yes
+      delegate_to: localhost
+
+    - set_fact:
+        ts: "{{lookup('pipe','date +%Y%m%d%H%M%S')}}"
+
+    - name: "Terminate the EC2 instance"
```

Comment:
A thought for a future version of this playbook: is there a way to do some sort of `try...except...finally` pattern here so that if this playbook or any nested playbook fails, the process will clean up the related AWS resources? I'd say this is low priority, since we're likely to be watching any process we kick off using these tools closely, at least initially. Might be a good candidate for the backlog.